### PR TITLE
[bugfix] Add capability to validate IPv4 and IPv6 addresses in [aci_dhcp_relay_policy]

### PR DESCRIPTION
**Pull request by shrsr**

### CODE REVIEW

File: aci/resource_aci_dhcprelayp.go [hunk 1]
```diff
@@ -81,16 +83,44 @@ func resourceAciDHCPRelayPolicy() *schema.Resource {
...
 				},
 			},
 		}),
 	}
 }
+
+func validateIPAddress() schema.SchemaValidateFunc {
+	return func(input interface{}, attribute string) (output []string, err []error) {
+		ip_address_input := input.(string)
+		ip := net.ParseIP("0.0.0.0")
```
Comment (lhercot) [inline aci/resource_aci_dhcprelayp.go:110]: You should be able to initialize it as:
```
var ip []bytes
```

or maybe:
```
var ip net.IP
```

File: aci/resource_aci_dhcprelayp.go [hunk 2]
```diff
@@ -81,16 +83,44 @@ func resourceAciDHCPRelayPolicy() *schema.Resource {
...
+								splitVal := strings.Split(val.(string), "/")
+								if len(splitVal) <= 1 {
+									ip := net.ParseIP(val.(string))
+									return ip.String()
+								} else {
+									ip := net.ParseIP(splitVal[0])
+									return ip.String() + "/" + splitVal[1]
+								}
+							},
+							ValidateFunc: schema.SchemaValidateFunc(validateIPAddress()),
```
Comment (lhercot) [inline aci/resource_aci_dhcprelayp.go:98]: Have you tried: validation.IsIPAddress ?

File: aci/resource_aci_dhcprelayp.go [hunk 3]
```diff
@@ -81,16 +83,44 @@ func resourceAciDHCPRelayPolicy() *schema.Resource {
 							ValidateFunc: validation.StringIsNotEmpty,
 						},
 						"addr": {
-							Type:         schema.TypeString,
-							Required:     true,
-							ValidateFunc: validation.IsIPv4Address,
+							Type:     schema.TypeString,
+							Required: true,
+							StateFunc: func(val interface{}) string {
+								splitVal := strings.Split(val.(string), "/")
```
Comment (lhercot) [inline aci/resource_aci_dhcprelayp.go:89]: The input here shouldn't have a netmask so not sure we actually need this split and if/else.

**Review by lhercot — CHANGES_REQUESTED** [review]

File: aci/resource_aci_dhcprelayp.go (hunk 1 above)
Comment (shrsr) [reply aci/resource_aci_dhcprelayp.go]: Removed this part

File: aci/resource_aci_dhcprelayp.go (hunk 2 above)
Comment (shrsr) [reply aci/resource_aci_dhcprelayp.go]: Yes, replaced the custom function with the inbuilt one

File: aci/resource_aci_dhcprelayp.go (hunk 3 above)
Comment (shrsr) [reply aci/resource_aci_dhcprelayp.go]: ok

**Review by lhercot — APPROVED** [review]
LGTM

**Pull request merged**